spyder_kernels/customize/spyderpdb.py

```diff
@@ -644,9 +644,14 @@ def precmd(self, line):
 
     def postcmd(self, stop, line):
         """
-        Notify spyder on any pdb command.
+        Notify spyder about a changed frame
         """
-        self.notify_spyder()
+        line = line.strip()
+        if line.startswith("!"):
+            line = line[1:]
+        cmd, arg, line = self.parseline(line.strip())
+        if cmd in ("up", "down", "jump", "u", "d", "j"):
```

Comment:
It is a good idea to check wether the update is needed!

I am not sure I am a fan of the list of commands. This duplicates a part of the `line` processing and the list might be subject to change. Why not check if the frame has changed instead, which is what we want to know?

My proposition would be to move the logic inside `notify_spyder`, where the filename and line number can be compared to the previous update, saved in a variable (`self._previous_step` maybe?).
Then `self._previous_step = None` in `postloop` so the next `preloop` would send an update as expected.

Also maybe `notify_spyder` could be renamed now that it only sends the debugger position?